apps/src/templates/instructions/TopInstructions.jsx

```diff
@@ -1,161 +1,31 @@
-  },
-  header: {
-    height: HEADER_HEIGHT,
-    lineHeight: HEADER_HEIGHT + 'px',
-    fontFamily: '"Gotham 4r"',
-    backgroundColor: color.lighter_purple,
-    textAlign: 'center'
-  },
-  body: {
-    backgroundColor: 'white',
-    overflowY: 'scroll',
-    paddingLeft: 10,
-    paddingRight: 10,
-    position: 'absolute',
-    top: HEADER_HEIGHT,
-    bottom: 0,
-    left: 0,
-    right: 0
-  },
-  embedView: {
-    height: undefined,
-    bottom: 0,
-    // Visualization is hard-coded on embed levels. Do the same for instructions position
-    left: 340
-  }
-};
-
-var TopInstructions = React.createClass({
+import TopInstructionsCSF from './TopInstructionsCSF';
+import TopInstructionsCSP from './TopInstructionsCSP';
+
+/**
+ * Simple class allowing us to have differences between CSF and CSP top
+ * instructions. Ultimately those might be combined back into one.
+ */
+const TopInstructions = React.createClass({
   propTypes: {
-    isEmbedView: React.PropTypes.bool.isRequired,
-    puzzleNumber: React.PropTypes.number.isRequired,
-    stageTotal: React.PropTypes.number.isRequired,
-    height: React.PropTypes.number.isRequired,
-    maxHeight: React.PropTypes.number.isRequired,
-    markdown: React.PropTypes.string,
-    collapsed: React.PropTypes.bool.isRequired,
-    toggleInstructionsCollapsed: React.PropTypes.func.isRequired,
-    setInstructionsHeight: React.PropTypes.func.isRequired,
-    onLoadImage: React.PropTypes.func.isRequired
-  },
-
-  /**
-   * Called externally
-   * @returns {number} The height of the rendered contents in pixels
-   */
-  getContentHeight: function () {
-    var instructionsContent = this.refs.instructions.refs.instructionsMarkdown;
-    return $(ReactDOM.findDOMNode(instructionsContent)).outerHeight(true);
+    isCsf: React.PropTypes.bool.isRequired
   },
-
-  /**
-   * Given a prospective delta, determines how much we can actually change the
-   * height (accounting for min/max) and changes height by that much.
-   * @param {number} delta
-   * @returns {number} How much we actually changed
-   */
-  onHeightResize: function (delta) {
-    var minHeight = MIN_HEIGHT;
-    var currentHeight = this.props.height;
-
-    var newHeight = Math.max(minHeight, currentHeight + delta);
-    newHeight = Math.min(newHeight, this.props.maxHeight);
-
-    this.props.setInstructionsHeight(newHeight);
-    return newHeight - currentHeight;
+  getRenderedHeight() {
+    return this.refs.topInstructions.getWrappedInstance().getRenderedHeight();
   },
 
-  componentDidMount: function () {
-    // Parent needs to readjust some sizing after images have loaded
-    $(ReactDOM.findDOMNode(this)).find('img').load(this.props.onLoadImage);
+  getCollapsedHeight() {
+    return this.refs.topInstructions.getWrappedInstance().getCollapsedHeight();
   },
 
-  render: function () {
-    if (!this.props.markdown) {
-      return <div/>;
-    }
-    var id = this.props.id;
+  render() {
+    const { props } = this;
```

Comment:
Equivalent to `const props = this.props`; Might be clearer to just do that in this cas.e